app/bundles/LeadBundle/Controller/AjaxController.php

```diff
@@ -894,4 +894,23 @@ private function prepareJsonResponse(int $leadCount): array
             'leadCount' => $leadCount,
         ];
     }
+
+    public function removeTagFromLeadAction(Request $request): JsonResponse
+    {
+        $dataArray = ['success' => 0];
+        $leadId    = (int) $request->request->get('leadId');
+        $tagId     = (int) $request->request->get('tagId');
+
+        if (!empty($leadId) && !empty($tagId)) {
+            /** @var \Mautic\LeadBundle\Model\LeadModel $leadModel */
+            $leadModel = $this->getModel('lead.lead');
+            \assert($leadModel instanceof LeadModel);
+
+            $leadModel->removeTagFromLead($leadId, $tagId);
+
+            $dataArray['success'] = 1;
+        }
+
+        return $this->sendJsonResponse($dataArray);
+    }
```

Comment:
Can be simplified with dependency injection:
```suggestion
    public function removeTagFromLeadAction(Request $request, LeadModel $leadModel): JsonResponse
    {
        $leadId    = (int) $request->request->get('leadId');
        $tagId     = (int) $request->request->get('tagId');

        if (!empty($leadId) && !empty($tagId)) {
            $leadModel->removeTagFromLead($leadId, $tagId);
        }

        return $this->sendJsonResponse([]);
    }
```
The success param is not checked anywhere so let's not set it. If needed, use HTTP codes instead.